text/0000-engines-component-dependencies.md

```diff
@@ -0,0 +1,97 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes to allow the declaration of components as dependencies of engines.
+
+# Motivation
+
+Currently, the only way to allow engines to share components with their consuming application/engine is to extract the component and package it as a standalone add-on. The add-on can then be installed by any application/engine that requires it.
+
+While this approach is a perfectly viable option in some circumstances, the viability begins to breakdown in distributed applications where the developer may not have control of some or all of the engines being consumed.
+
+In such circumstances it becomes difficult to ensure consistency throughout the application as each engine could theoretically be using a different version of the same component, potentially resulting in different areas of the application having different UI/UX for the same component/functionality.
```

Comment:
I'm not 100% sure this is always a drawback.

Consistent UI could theoretically be handled by some CSS tricks and the differences in UX might be required. For instance, if you built your Engine with the assumption that a component opens a modal and then all of a sudden that component now open an inline form, it has the potential to break other aspects of your UX.

One thing I would like to see highlighted in this doc is that when using addons, you are guaranteed that the Engine is getting the exact component it wants.